homeassistant/components/squeezebox/config_flow.py

```diff
@@ -222,3 +242,55 @@ async def async_step_dhcp(
 
         # if the player is unknown, then we likely need to configure its server
         return await self.async_step_user()
+
+
+class OptionsFlowHandler(OptionsFlow):
+    """Options Flow Handler."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Options Flow Steps."""
+
+        errors: dict[str, str] = {}
+
+        async def _validate_input(data: dict[str, Any]) -> str | None:
+            """Validate the user input allows us to connect."""
+
+            if data[CONF_BROWSE_LIMIT] < 1 or data[CONF_VOLUME_STEP] < 1:
+                raise ValueError
+
+            return None
```

Comment:
Why do make this a nested function?
I think it would be more readable as a static class function or even as a module function.

Reply:
Or even better drop this validation routine and use a number selector instead with a min value

Reply:
Yes, to be consistent at least with the other validate functions in the file, it shouldn't be nested there.  I'll move it.

Reply:
You can also look at `vol.All(vol.Coerce(int), vol.Range(min=1, max=65535))` if you don't want to use a Selector (though I think Selectors are the way forward nowadays)

Reply:
Interesting - I'm not aware of how to use selectors on this context.  Could you point me at an example?

Reply:
Something like this should do it:
```
        vol.Required(CONF_BROWSE_LIMIT): vol.All(
            selector.NumberSelector(
                selector.NumberSelectorConfig(
                    min=1, max=XYZ, mode=selector.NumberSelectorMode.BOX
                )
            ),
            vol.Coerce(int),
        ),
        vol.Required(CONF_VOLUME_STEP): vol.All(
            selector.NumberSelector(
                selector.NumberSelectorConfig(
                    min=1, max=XYZ, mode=selector.NumberSelectorMode.BOX
                )
            ),
            vol.Coerce(int),
        ),
```

Reply:
nice - thanks - I had no idea you could use selectors in the config.  I'll update that.

Reply:
done